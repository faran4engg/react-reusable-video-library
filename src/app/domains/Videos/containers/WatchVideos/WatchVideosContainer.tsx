import { FC } from 'react';
import { useRouteMatch } from 'react-router-dom';

import { MatchParams, OwnProps, RenderProps } from './types';

import { useApprovedVideos } from 'app/hooks/queries-hooks';

const WatchVideosContainer: FC<OwnProps & RenderProps> = ({ children }) => {
  const match = useRouteMatch<MatchParams>();
  const { id } = match.params;

  const { data: approvedVideos } = useApprovedVideos();

  return children({
    id,
    currentlyWatching: approvedVideos?.data[id],
    approvedVideos,
  });
};

export default WatchVideosContainer;
