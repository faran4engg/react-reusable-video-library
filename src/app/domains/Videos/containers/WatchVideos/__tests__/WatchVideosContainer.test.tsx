import React from 'react';
// eslint-disable-next-line import/no-extraneous-dependencies
import routeData from 'react-router';
import { createRenderer } from 'react-test-renderer/shallow';
import WatchVideosContainer from '..';

const mockMatch = {
  params: { id: 'video-id' },
  isExact: false,
  path: 'asd',
  url: 'adas',
};
beforeEach(() => {
  jest.spyOn(routeData, 'useRouteMatch').mockReturnValue(mockMatch);
});

describe('<WatchVideosContainer />', () => {
  afterEach(() => {
    jest.clearAllMocks();
  });
  it('should render and match the snapshot when loading', () => {
    const renderer = createRenderer();

    renderer.render(
      <WatchVideosContainer>{() => <div />}</WatchVideosContainer>,
    );
    const renderedOutput = renderer.getRenderOutput();
    expect(renderedOutput).toMatchSnapshot();
  });
});
