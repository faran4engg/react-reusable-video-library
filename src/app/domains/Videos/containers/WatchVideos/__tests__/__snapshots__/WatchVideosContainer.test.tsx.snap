// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WatchVideosContainer /> should render and match the snapshot when loading 1`] = `<div />`;
